gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/SSOCookieFederationFilter.java

```diff
@@ -146,11 +146,11 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
     if (ssoCookies.isEmpty()) {
       /* check for unauthenticated paths to bypass */
       for (final String path : unAuthenticatedPaths) {
-        if (req.getRequestURI().contains(path)) {
+        if (req.getPathInfo().equals(path)) {
```

Comment:
Could unAuthenticatedPaths.contains(req.getPathInfo()) be employed here? Should this logic go into a util method, which could be shared across all these implementations?

Reply:
yes, it should, I will refactor the code.